spec/support/constraints.rb

```diff
@@ -97,11 +91,77 @@ def require_ssl
     end
   end
 
+  def require_local_tls
+    before do
+      unless SpecConfig.instance.ssl? && !SpecConfig.instance.ci?
+        skip 'Not running locally with TLS enabled'
+      end
+    end
+  end
+
   def require_no_retry_writes
     before do
       if SpecConfig.instance.retry_writes?
         skip "Retry writes is enabled"
       end
     end
   end
+
+  def require_compression
+    before do
+      if SpecConfig.instance.compressors.nil?
+        skip "Compression is not enabled"
+      end
+    end
+  end
+
+  def require_no_compression
+    before do
+      if SpecConfig.instance.compressors
+        skip "Compression is enabled"
+      end
+    end
+  end
+
+  def min_ruby_version(version)
+    before do
+      if RUBY_VERSION < version
+        skip "Ruby version #{version} or higher required"
+      end
+    end
+  end
+
+  def ruby_version_lt(version)
```

Comment:
Can we change the name of either this or `min_ruby_version`? They're both doing the inverse of each other, so I think it would make sense to either rename this something like `max_ruby_version_exclusive` or to rename `min_ruby_version` `ruby_version_gle`.

Reply:
Changed to ruby_version_gte.